charts/kube-master/templates/dex.yaml

```diff
@@ -0,0 +1,264 @@
+{{ if and .Values.dashboard .Values.dashboard.enabled }}
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+  name: {{ include "master.fullname" . }}-dex 
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ include "master.fullname" . }}-dex 
+      release: {{ .Release.Name }}  
+  template:
+    metadata:
+      labels:
+        app: {{ include "master.fullname" . }}-dex 
+        release: {{ .Release.Name }} 
+    spec:
+      initContainers:
+      - image: xmartlabs/htpasswd 
```

Comment:
Extra points for going the extra mile to avoid leaking the secret in the ConfigMap. ❤️ 

But I think this is an awful lot of yaml and a basically untrusted container we add here.
So let me provide an alternative suggestion without an init container:
* I would generate the bcrypt hash in the operator when the helm values are computed. This frees us from  the `htpasswd` dependency during runtime
* Then I see three options where to put the password:
1. but the whole config in a secret instead of a ConfigMap and mount that
2. leave it in a ConfigMap and do the SED magic in the main container before starting dex
3. but the hashed password in the configmap and rely on bcrypt being reasonably safe from decryption

All three options are fine by me and I guess the third one would be the simplest with an arguably negligible decrease in security.

Reply:
In the third option, if you are suggesting the following, I can implement and test it: 
- Hashing the `DashboardStaticPassword` from the secret in `KlusterToHelmValues` and passing the hashed value as to the `kube-master` chart.

Should we also pass `DashboardClientSecret` as raw Helm value? It is also replaced by `sed` in the init container and there is no hashing for it.

Reply:
It is implemented with https://github.com/sapcc/kubernikus/pull/464/commits/8bd00d13b31c83eddd3a01424bff555b14465b38